bob-common/src/metrics/mod.rs

```diff
@@ -111,16 +118,51 @@ pub type Timer = Instant;
 /// Structure contains put/get metrics for `BobClient`
 #[derive(Debug, Clone)]
 pub struct BobClient {
-    prefix: String,
+    prefixed_names: Arc<PrefixedNames>,
+}
+
+#[derive(Debug, Clone)]
+struct PrefixedNames {
+    put_count: String,
+    put_timer: String,
+    put_error_count: String,
+    get_count: String,
+    get_timer: String,
+    get_error_count: String,
+    exist_count: String,
+    exist_timer: String,
+    exist_error_count: String,
+    delete_count: String,
+    delete_timer: String,
+    delete_error_count: String,
+}
+
+impl PrefixedNames {
+    fn new(prefix: &str) -> Self {
+        Self {
+            put_count: format!("{}.put_count", prefix),
+            put_timer: format!("{}.put_timer", prefix),
+            put_error_count: format!("{}.put_error_count", prefix),
+            get_count: format!("{}.get_count", prefix),
+            get_timer: format!("{}.get_timer", prefix),
+            get_error_count: format!("{}.get_error_count", prefix),
+            exist_count: format!("{}.exist_count", prefix),
+            exist_timer: format!("{}.exist_timer", prefix),
+            exist_error_count: format!("{}.exist_error_count", prefix),
+            delete_count: format!("{}.delete_count", prefix),
+            delete_timer: format!("{}.delete_timer", prefix),
+            delete_error_count: format!("{}.delete_error_count", prefix),
+        }
+    }
 }
 
 impl BobClient {
-    fn new(prefix: String) -> Self {
-        BobClient { prefix }
+    fn new(prefixed_names: Arc<PrefixedNames>) -> Self {
+        BobClient { prefixed_names }
     }
 
     pub(crate) fn put_count(&self) {
-        counter!(self.prefix.clone() + ".put_count", 1);
+        counter!(self.prefixed_names.put_count.clone(), 1);
```

Comment:
It will requre `'static` strings to construct, and they can be either compile-time defined or cloned

Reply:
It has conversion from `String`: https://docs.rs/metrics/0.20.1/metrics/struct.KeyName.html#method.from

Reply:
It will require updating metrics crate to `0.18.0` version, which breaks some code (method changes). I think it's better to make metrics update separate ticket

Reply:
Agree with that. Created a separate issue: https://github.com/qoollo/bob/issues/628